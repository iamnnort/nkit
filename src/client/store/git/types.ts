import { CommonStore } from '../../lib/store/types';

export enum ActionTypes {
  LOAD_RELEASES_REQUEST = '@@git/LOAD_RELEASES_REQUEST',
  LOAD_RELEASES_SUCCESS = '@@git/LOAD_RELEASES_SUCCESS',
  LOAD_RELEASES_FAILURE = '@@git/LOAD_RELEASES_FAILURE',
}

export interface Git extends CommonStore {
  releases: GitRelease[];
}

export interface GitRelease {
  id: number;
  name: string;
  tag_name: string;
  html_url: string;
}
