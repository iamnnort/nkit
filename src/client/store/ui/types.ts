export enum ActionTypes {
  TOGGLE_SIDEBAR = '@@ui/TOGGLE_SIDEBAR',
}

export interface UiInterface {
  isSidebarOpened: boolean;
}
