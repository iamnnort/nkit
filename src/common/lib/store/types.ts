export interface CommonStore {
  isInitialLoaded: boolean;
  isLoading: boolean;
  error: string;
}

export enum CommonError {
  FAILED_REQUEST = 'Loading was failed',
}
