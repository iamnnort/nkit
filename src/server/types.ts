import * as express from 'express';
import i18next from 'i18next';

export interface Request extends express.Request {
  i18n: i18next.i18n;
}
