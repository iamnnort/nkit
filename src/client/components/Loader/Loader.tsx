import * as React from 'react';

import { ComponentProps } from './Loader.types';
import { Loader, LoaderSpin } from './Loader.styled';

const LoaderComponent: React.FC<ComponentProps> = ({ isLoading, render }) => (
  <Loader>{isLoading ? <LoaderSpin /> : render()}</Loader>
);

export default LoaderComponent;
