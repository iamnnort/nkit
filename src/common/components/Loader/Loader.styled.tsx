import styled from '@common/theme/styled-components';

export const LoaderSpin = styled.div`
  height: 33px;

  margin: 15px 0;

  background: url('${require('./loader.gif')}') no-repeat center;
  background-size: 33px;
`;

export const LoaderError = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;

  height: 33px;

  margin: 15px 0;

  font-size: 12px;
  color: ${(props) => props.theme.errorColor};
`;
