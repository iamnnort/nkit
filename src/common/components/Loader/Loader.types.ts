export interface ComponentProps {
  isInitialLoaded: boolean;
  isLoading: boolean;
  error: string;
  render: () => React.ReactNode;
}
