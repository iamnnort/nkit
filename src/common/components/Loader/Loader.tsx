import * as React from 'react';

import { ComponentProps } from './Loader.types';
import { LoaderSpin, LoaderError } from './Loader.styled';

const LoaderComponent: React.FC<ComponentProps> = ({ isLoading, isInitialLoaded, error, render }) => {
  const Content = isInitialLoaded && render();
  const Error = error && <LoaderError>{error}</LoaderError>;

  return isLoading ? (
    <LoaderSpin />
  ) : (
    <React.Fragment>
      {Content}
      {Error}
    </React.Fragment>
  );
};

export default LoaderComponent;
