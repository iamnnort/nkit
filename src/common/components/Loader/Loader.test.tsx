import * as React from 'react';
import { mount, ReactWrapper } from 'enzyme';

import LoaderComponent from '@common/components/Loader/Loader';
import { ComponentProps } from '@common/components/Loader/Loader.types';
import { LoaderError, LoaderSpin } from '@common/components/Loader/Loader.styled';

describe('LoaderComponent', () => {
  let wrapper: ReactWrapper;

  const props: ComponentProps = {
    isInitialLoaded: true,
    isLoading: false,
    error: '',
    render: () => <div id='fakeContent'>Fake Content</div>,
  };

  beforeEach(() => {
    wrapper = mount(<LoaderComponent {...props} />);
  });

  it(`should not regret`, () => {
    expect(wrapper).toMatchSnapshot();
  });

  it(`should render content if it was loaded`, () => {
    expect(wrapper.exists('#fakeContent')).toBe(true);
  });

  it(`should render the error if it was loaded`, () => {
    wrapper.setProps({
      error: 'Fake error',
    });

    expect(wrapper.exists(LoaderError)).toBe(true);
  });

  it(`should render a loader spin if it is still loading`, () => {
    wrapper.setProps({
      isLoading: true,
    });

    expect(wrapper.exists(LoaderSpin)).toBe(true);
  });
});
