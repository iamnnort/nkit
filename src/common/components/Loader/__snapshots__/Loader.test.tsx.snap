// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderComponent should not regret 1`] = `
<LoaderComponent
  error=""
  isInitialLoaded={true}
  isLoading={false}
  render={[Function]}
>
  <div
    id="fakeContent"
  >
    Fake Content
  </div>
</LoaderComponent>
`;
